src/components/tabs/tab-bar.tsx

```diff
@@ -1,131 +1,175 @@
+          previousElement.style.marginRight = `${elemn.getBoundingClientRect().width - 1}px`;
+        } else {
+          nextElement.style.marginLeft = `${elemn.getBoundingClientRect().width - 1}px`;
+        }
       }
-      tabBar.current.onmouseup = (e: any) => closeDragElement(e, e.target);
     }
   }
 
-  function elementDrag(e: any, elemn: any) {
-    // drags the tab only on y axis,
-    // alongside with a "ghost tab" that will preview the next position
-    e = e || window.event;
-      // calculate the new cursor position:
-    pos1.current = pos3.current - e.clientX;
-    pos3.current = e.clientX;
-      // set the element's new position:
-    const exactPos = elemn.offsetLeft - pos1.current;
-    elemn.style.left = `${exactPos}px`;
-    const sibling = elemn.nextSibling as HTMLElement;
-    const newElement = document.getElementById('new');
-      // if the tab is pushing foward
-    if (sibling && (sibling.offsetLeft - 80) < exactPos) {
-      const sibling = elemn.nextSibling as HTMLElement;
-      if (sibling.id !== 'new') {
-        sibling.className = 'animated';
-      }
-      tabBar.current.insertBefore(sibling, elemn);
-      tabBar.current.insertBefore(newElement, sibling.nextSibling);
-    } else {
-        // if the tab is pushing backwards
-      const sibling = elemn.previousSibling &&
-                        elemn.previousSibling.previousSibling as HTMLElement;
-      if (sibling && sibling.offsetLeft + 90 > exactPos) {
-        if (sibling.id !== 'new') {
-          sibling.className = 'deanimated';
+  function elementDrag(e: React.MouseEvent<HTMLElement>) {
+    if (dragged) {
+      const position = exactPos(e);
+      const currentElement = getRef(dragged).current;
+      const nextElement = currentElement.nextSibling as HTMLElement;
+      const previousElement = currentElement.previousSibling as HTMLElement;
+      currentElement.style.left = `${position}px`;
+      if (nextElement && nextElement.getBoundingClientRect().left - 70 < position) {
+        if (previousElement) {
+          previousElement.style.marginRight = '0px';
+          previousElement.style.marginLeft = '0px';
         }
-        tabBar.current.insertBefore(newElement, sibling);
-        tabBar.current.insertBefore(elemn, sibling);
+        nextElement.style.marginLeft = '0px';
+        nextElement.style.marginRight = `${currentElement.getBoundingClientRect().width - 1}px`;
+        nextElement.className = 'animated';
+        array_move(tabList, tabList.indexOf(dragged), tabList.indexOf(dragged) + 1);
+        setTabList([...tabList]);
+      } else if (previousElement && previousElement.getBoundingClientRect().right - 80 > position) {
+        if (nextElement) {
+          nextElement.style.marginRight = '0px';
+          nextElement.style.marginLeft = '0px';
+        }
+        previousElement.style.marginRight = '0px';
+        previousElement.style.marginLeft = `${currentElement.getBoundingClientRect().width - 1}px`;
```

Comment:
Please create a variable for this expression, describing what this expression means
`const widthForSomething = currentElement.getBoundingClientRect().width - 1`

And when you use it, it will be more meaningful, like this:

> blabla.width = `${widthForSomething}px`